transform/binary-plugin/binary_plugin.go

```diff
@@ -14,14 +14,40 @@ import (
 
 type BinaryPlugin struct {
 	commandRunner
-	log logrus.FieldLogger
+	pluginMetadata transform.PluginMetadata
+	log            logrus.FieldLogger
 }
 
-func NewBinaryPlugin(path string) transform.Plugin {
-	return &BinaryPlugin{commandRunner: &binaryRunner{pluginPath: path}, log: logrus.New().WithField("pluginPath", path)}
+// NewBinaryPlugin -
+func NewBinaryPlugin(path string) (transform.Plugin, error) {
```

Comment:
there is some logic being added here, would it be fair to ask for a unit?

Reply:
Added